ipamd/ipamd.go

```diff
@@ -194,10 +201,19 @@ func New(k8sapiClient k8sapi.K8SAPIs, eniConfig *eniconfig.ENIConfigController)
 func (c *IPAMContext) nodeInit() error {
 	ipamdActionsInprogress.WithLabelValues("nodeInit").Add(float64(1))
 	defer ipamdActionsInprogress.WithLabelValues("nodeInit").Sub(float64(1))
-	maxENIs, err := c.awsClient.GetENILimit()
-	if err == nil {
+
+	// If MAX_ENI is set, the maximum number of ENIs may be no more than that
+	// number (but might be less, depending on instance type.) If MAX_ENI is < 1,
+	// then we use the limit from the instance type.
+	maxENIs := getMaxENI()
```

Comment:
If `getMaxENI()` took `instanceENILimit` as an argument, the limit checks could be in once place and the check would just be something like
```go
	instanceMaxENIs, err := c.awsClient.GetENILimit()
	if err == nil {
		maxENIs = getMaxENI(instanceMaxENIs)
	}
```